dynosaur_derive/src/expand.rs

```diff
@@ -148,6 +138,19 @@ pub(crate) fn expand_sig_ret_ty_to_pin_box(sig: &mut Signature) {
     sig.output = parse_quote! { #arrow ::core::pin::Pin<Box<dyn #ret + 'dynosaur>> };
 }
 
+pub(crate) fn expand_sig_ret_ty_to_box(sig: &mut Signature) {
+    let (arrow, ret) = expand_arrow_ret_ty(sig);
+    if let Some(asyncness) = sig.asyncness.take() {
+        sig.fn_token.span = asyncness.span;
```

Comment:
I've seen this in a few places in the code and don't understand why we do it.

Reply:
Given `async fn foo()`, we do `asyncness.take()` to remove the async part of the fn. Now we have `fn foo() -> impl Future<Output=()>` and effectively the span of the fn keyword in the new function is the span of the async keyword of the old function.

Reply:
Why not make it the span of the `fn` keyword of the old function?

Reply:
We should try to address this in a followup as this is preexisting anyway. We're doing the same in `expand_sig_ret_ty_to_rpit`

Reply:
Doing it in a followup is fine